src/components/buttons/Button.spec.jsx

```diff
@@ -54,6 +54,25 @@ test('icon', () => {
   expect(button.find('.sg-button__icon')).toHaveLength(1);
 });
 
+test('icon only', () => {
+  const icon = <span>:P</span>;
+  const button = shallow(
+    <Button icon={icon} iconOnly>
+      Some text
+    </Button>
+  );
+
+  expect(button.contains(icon)).toEqual(true);
+  expect(button.find('.sg-button--icon-only')).toHaveLength(1);
+  expect(button.find('.sg-button__icon')).toHaveLength(1);
```

Comment:
maybe add additional expect to check that the text is not rendered?

Reply:
but children are optional already, if you pass some text i will be rendered but not visible